### PR TITLE
[Pipeline] Add zero shot audio classificatoin pipeline

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
Add the `zero_shot_audio_classification_pipeline`  for the `CLAP` models. See #21370

### CODE REVIEW

File: src/transformers/pipelines/zero_shot_audio_classification.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    provide an audio and a set of `candidate_labels`.
+
+    Example:
+    ```python
+    >>> from transformers import pipeline
+    >>> from datasets import load_dataset
+
+    >>> dataset = load_dataset("ashraq/esc50")
+    >>> audio = next(iter(dataset["train"]["audio"]))["array"]
+    >>> classifier = pipeline(task="zero-shot-audio-classification", model="laion-ai/clap-hsat-tiny")
```
Comment (younesbelkada) [inline src/transformers/pipelines/zero_shot_audio_classification.py:35]:
```suggestion
    >>> classifier = pipeline(task="zero-shot-audio-classification", model="laion-ai/clap-htsat-tiny")
```
it's `htsat` no?

File: src/transformers/pipelines/zero_shot_audio_classification.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    >>> classifier(
+    ...     audio,
+    ...     candidate_labels=["Sound of a dog", "Sound of vaccum cleaner"],
+    ... )
+    [{'score': 0.999727189540863, 'label': 'Sound of a dog'}, {'score': 0.0002727957325987518, 'label': 'Sound of vaccum cleaner'}]
```
Comment (younesbelkada) [inline src/transformers/pipelines/zero_shot_audio_classification.py:40]:
```suggestion
    ... )
    [{'score': 0.999727189540863, 'label': 'Sound of a dog'}, {'score': 0.0002727957325987518, 'label': 'Sound of vaccum cleaner'}]
```
Need to check the score otherwise the doc-test (if any) won't probably pass?

File: src/transformers/pipelines/zero_shot_audio_classification.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    [huggingface.co/models](https://huggingface.co/models?filter=zero-shot-audio-classification).
+    """
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+        if self.framework != "pt":
+            raise ValueError(f"The {self.__class__} is only available in PyTorch.")
+        # No specific FOR_XXX available yet
+        # self.check_model_type(MODEL_FOR_IMAGE_CLASSIFICATION_MAPPING)
```
Comment (younesbelkada) [inline src/transformers/pipelines/zero_shot_audio_classification.py:56]:
```suggestion
```

**Review by younesbelkada — APPROVED** [review]
Thanks for your great work! Left few nits - 
let's wait for @Narsil 's approval before merging

File: src/transformers/pipelines/zero_shot_audio_classification.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+    Example:
+    ```python
+    >>> from transformers import pipeline
+    >>> from datasets import load_dataset
+
+    >>> dataset = load_dataset("ashraq/esc50")
+    >>> audio = next(iter(dataset["train"]["audio"]))["array"]
+    >>> classifier = pipeline(task="zero-shot-audio-classification", model="laion-ai/clap-htsat-tiny")
+    >>> classifier(
```
Comment (younesbelkada) [inline src/transformers/pipelines/zero_shot_audio_classification.py:36]:
```suggestion
    >>> pipeline_output = classifier(
```
Something like that combined with the suggestion below should do the trick too

**Review by Narsil — APPROVED** [review]

File: src/transformers/pipelines/base.py [hunk 5]
```diff
@@ -77,7 +77,7 @@ def _pad(items, key, padding_value, padding_side):
         # Others include `attention_mask` etc...
         shape = items[0][key].shape
         dim = len(shape)
-        if key in ["pixel_values", "image"]:
+        if key in ["pixel_values", "image", "input_features"]:
```
Comment (ArthurZucker) [inline src/transformers/pipelines/base.py:80]: Failing test might be related ot this! Will investigate

Comment (ArthurZucker) [reply src/transformers/pipelines/base.py:80]: Indeed, fixing this then will be good to go! 🔥

**Comment by Narsil** [comment]
LGTM !

File: src/transformers/pipelines/zero_shot_audio_classification.py [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+
+        if self.framework != "pt":
+            raise ValueError(f"The {self.__class__} is only available in PyTorch.")
+        # No specific FOR_XXX available yet
+
+    def __call__(
+        self,
+        audios: Union[np.ndarray, bytes, str],
+        **kwargs,
+    ):
```
Comment (sgugger) [inline src/transformers/pipelines/zero_shot_audio_classification.py:57]: Looks like this fits in one line.

File: src/transformers/pipelines/zero_shot_audio_classification.py [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+        """
+        Args:
+        Assign labels to the audio(s) passed as inputs.
```
Comment (sgugger) [inline src/transformers/pipelines/zero_shot_audio_classification.py:60]: The args should be below the description.

**Review by sgugger — APPROVED** [review]
Thanks for working on this!
